### PR TITLE
Update overview product when user clicks on related product

### PR DESCRIPTION
**Pull request by cschucode**
This PR does the following:
* Uses state to update product id when user clicks on related products card.
* Refactors where the rating is coming from on cards
* Refactors where image URL comes from on cards

**[IMPORTANT]**
The handleClick updates the product id and current product which will affect both
* Product Overview 
* Review Rating
but hopefully in a way that's useful.

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -16,27 +16,41 @@ const App = () => {
...
+  };
 
   return (
     <div className="app__container">
       <header className="app__header">Logo and Search Go Here</header>
-      {isLoading ? <p>Loading!</p> : <ProductOverview productId={currentProduct.id} />}
-      {isLoading ? <p>Loading!</p>
-        : <RelatedProducts rating={rating} currentProduct={currentProduct} />}
-      {currentProduct && <ReviewRating id={currentProduct.id} />}
+      {!isLoading ? (
```
Comment (cschucode) [inline client/components/App.jsx:47]: All three of our components were waiting on the load from the API, so I combined them to make it cleaner.

File: client/components/ProductOverview/ProductOverview.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import ImageGallery from './ImageGallery';
 
 import '../../css/ProductOverview.scss';
 
-const ProductOverview = ({ productId }) => {
+const ProductOverview = ({ currentProduct, productId }) => {
```
Comment (cschucode) [inline client/components/ProductOverview/ProductOverview.jsx]: I passed in the current product just to see if it was updating when the user clicks on the related product card. Feel free to use/modify what you need for the remaining sections of the Product Overview.

File: client/tests/ProductOverview.test.jsx [hunk 3]
```diff
@@ -6,7 +6,19 @@ import { render, screen } from '@testing-library/react';
 import ProductOverview from '../components/ProductOverview/ProductOverview';
 
 beforeEach(() => {
-  render(<ProductOverview />);
+  const mockCurrent = {
```
Comment (cschucode) [inline client/tests/ProductOverview.test.jsx:9]: Updated these tests to handle required prop types.

File: client/components/relatedProducts/Card.jsx [hunk 4]
```diff
@@ -1,42 +1,57 @@
...
+  const hasNoRating = Number.isNaN(rating);
+
+  useEffect(() => {
+    getStarRatingAvg(relatedProduct.id).then((result) => setRating(result.avg));
+    getImageUrl(relatedProduct.id).then((url) => setImageUrl(url));
+  }, []);
+
+  return (
+    <div className="related-products__card" role="button" tabIndex={0} onKeyPress={() => { handleCardClick(relatedProduct.id); }} onClick={() => { handleCardClick(relatedProduct.id); }}>
+      <div className="related-products__card-image" style={imageUrl ? { backgroundImage: `url(${imageUrl})` } : {}}>
```
Comment (cschucode) [inline client/components/relatedProducts/Card.jsx:28]: This line uses a ternary to use the default background image in Card.scss when one isn't provided

**Review by runhui2010 — APPROVED** [review]
Nice update.

File: client/components/App.jsx (hunk 1 above)
Comment (jleiandy) [reply client/components/App.jsx:47]: Nice refactoring, I also handled the merge conflict with the axios call and the getStarRating component. getStarRating now passes both the average review and review count properly to components.

File: client/components/ProductOverview/ProductOverview.jsx (hunk 2 above)
Comment (jleiandy) [reply client/components/ProductOverview/ProductOverview.jsx]: I am currently making an axios call with the productId to get the main product. But I can now refactor to just use the currentProduct passed in, thanks.

File: client/tests/ProductOverview.test.jsx (hunk 3 above)
Comment (jleiandy) [reply client/tests/ProductOverview.test.jsx:9]: thank you for this, I should've done this already but I appreciate your help.

**Review by jleiandy — APPROVED** [review]
Overall looks good, I expect some merge conflicts in the App level and in my product overview component so we should discuss how to resolve those together or else we may end up reverting each other's changes.

**Pull request merged**